app/workers/email_archive_worker.rb

```diff
@@ -4,17 +4,20 @@ class EmailArchiveWorker
   sidekiq_options queue: :cleanup
 
   LOCK_NAME = "email_archive_worker".freeze
+  BATCH_SIZE = 1000
 
   def perform
     return unless ENV.include?("EMAIL_ARCHIVE_S3_ENABLED")
     Email.with_advisory_lock(LOCK_NAME, timeout_seconds: 0) do
       start_time = Time.zone.now
       archived_count = 0
 
-      EmailArchiveQuery.call.in_batches do |batch|
-        Email.transaction do
-          archived_count += archive_batch(batch.as_json)
-        end
+      loop do
+        batch = EmailArchiveQuery.call.limit(BATCH_SIZE).as_json
+
+        archived_count += archive_batch(batch)
```

Comment:
I think we might need something here (or in the `archive_batch` method) to deal with the scenario that the query returns 0 results.

Reply:
Good spot, I've added another `break`. Let me know what you think?